proofs/tree.go

```diff
@@ -237,6 +249,23 @@ type DocumentTree struct {
 	compactProperties bool
 }
 
+func (saltsMap InnerCompactsSaltsMap) innerSaltsMapToOuter () (CompactsSaltsMap){
```

Comment:
What exactly do you need inner & outer for?

Reply:
Using hex encoded strings inflates the key to 4x the size which I would prefer to avoid. An alternative would be to just use a slice of a simple struct with key & salt. Perhaps that would be easiest?

Reply:
OK

Reply:
It will do some harm to performance when searching salt for a compact in slice

Reply:
I'm not sure if the performance of a string map is actually better. Wouldn't it also just rely on an equality check? 

I think 4x the memory footprint might be worse than the performance hit we take. I'm not too worried about the performance at the moment.